code/backend/batch/utilities/helpers/AzureSearchHelper.py

```diff
@@ -31,9 +31,7 @@ def get_vector_store(self):
                 name="content_vector",
                 type=SearchFieldDataType.Collection(SearchFieldDataType.Single),
                 searchable=True,
-                vector_search_dimensions=len(
-                    llm_helper.get_embedding_model().embed_query("Text")
```

Comment:
I understand that we don't need to call `llm_helper.get_embedding_model().embed_query` if we already have that value in config. But we make 3 calls to that in this method anyway. Is it worth calling it once at the top anyway.

Would the call to that method every not return what was configured in AZURE_SEARCH_DIMENSIONS?

Reply:
We don't need to call this method at all, essentially what was happening before was that we were getting the embeddings for a dummy string, i.e. "Text", and using the length of the array of embeddings to calculate what the dimension size should be. But for this model, `text-embedding-ada-002` it's always `1536`, so we don't need to make a call to OpenAI, we can just use the value directly, as set in the env var.

https://learn.microsoft.com/en-us/azure/ai-services/openai/concepts/models#embeddings-models

Also, in the other places where `llm_helper.get_embedding_model().embed_query` is still present, we're not actually calling the function, we're just passing the function to `AzureSearch`

Reply:
What if a different embeddings model is used, will this change break that?

Reply:
If a different model would be used then they would have to change the value of the env var, I don't think that would be the worst thing in the world, however I've updated it now so that we call the embeddings once and then cache it in the class, so we don't need to calculate it again.

I'm not sure how much we're really saving here, since everytime we call `AzureSearch()` from langchain, that module does the exact same thing and gets the embeddings for `"Text"`, however I think the caching in the class (or something similar) is a pattern that we should look to put in multiple places to reduce latency in the app.